satpy/readers/gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""A reader for the Top of Atmosphere outgoing fluxes from the Geostationary Earth Radiation
+Budget instrument aboard the Meteosat Second Generation satellites."""
+
+
+import logging
+from datetime import timedelta
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+def gerb_get_dataset(hfile, name):
+    """
+    Load a GERB dataset in memory from a HDF5 file
+
+    The routine takes into account the quantisation factor and fill values.
+    """
+    ds = hfile[name]
+    if 'Quantisation Factor' in ds.attrs and 'Unit' in ds.attrs:
+        ds_real = ds[...]*ds.attrs['Quantisation Factor']
+    else:
+        ds_real = ds[...]*1.
+    ds_min = ds[...].min()
+    if ds_min < 0:
+        mask = ds == ds_min
+        ds_real[mask] = np.nan
+    return ds_real
+
+
+class GERB_HR_FileHandler(BaseFileHandler):
+    """File handler for GERB L2 High Resolution H5 files."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(GERB_HR_FileHandler, self).__init__(filename,
+                                                  filename_info,
+                                                  filetype_info)
+        self._h5fh = h5py.File(self.filename, 'r')
+        self.ssp_lon = self._h5fh["Geolocation"].attrs["Nominal Satellite Longitude (degrees)"][()]
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        return self.start_time + timedelta(minutes=14, seconds=59)
```

Comment:
Normally this `end_time` is the nominal time of the data and is usually pretty "human friendly". I'm guessing the repeat cycle of the instrument is 15 minutes? In that case could we make this 15 minutes instead of 14:59? Or in some cases we just make this equal to `start_time` if we can't get it from the file data.

Reply:
Would making the end time exactly 15 mins not present problems as it'll overlap with the start time of the next scan, though?

Reply:
@simonrp84 that was my motivation (plus the fact that some other readers seem to do that)

Reply:
@pdebuyl What other readers? I tried to find it when I made my comment, but I was pretty sure @mraspaud and I defined `end_time`s in Satpy as being exclusive. So start/end times being every 15 minutes would be fine because any downstream application should be interpreting the end time as happening the instant before the time specified.

Reply:
By chance, I had a look at the `hsaf_h5` reader (for inspiration as it is another HDF5 reader) and that one use 23 hours 59 minutes, hence my confusion. I'll put 15 minutes.

Reply:
It looks like that was a suggestion from @pnuu here:

https://github.com/pytroll/satpy/pull/2285#discussion_r1036763660

Perhaps we need to discuss this further. I know the FCI folks and I had trouble in our SIFT application when the nice human-friendly times weren't used because it was awkward to represent the blocks of time in a GUI...but maybe that was more about using real observation/scanning times rather than off-by-one inclusive times versus exclusive times.